internal/pkg/service/appsproxy/dns/client.go

```diff
@@ -0,0 +1,93 @@
+package dns
+
+import (
+	"context"
+	"fmt"
+	"math/rand"
+	"net"
+	"strings"
+	"time"
+
+	"github.com/miekg/dns"
+
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+const DialTimeout = 2 * time.Second
+
+const DNSReadTimeout = 2 * time.Second
+
+const DNSWriteTimeout = 2 * time.Second
+
+type Client struct {
+	client    *dns.Client
+	dnsServer string
+}
+
+func NewClientFromEtc(dialer *net.Dialer) (*Client, error) {
+	// Parse DNS config
+	dnsCfgFile := "/etc/resolv.conf"
+	dnsCfg, err := dns.ClientConfigFromFile(dnsCfgFile)
+	if err != nil {
+		return nil, err
+	}
+	if len(dnsCfg.Servers) == 0 {
+		return nil, errors.Errorf(`no DNS server found in "%s"`, dnsCfgFile)
+	}
+
+	// Get DNS server
+	dnsServer := dnsCfg.Servers[0]
+	if !strings.Contains(dnsServer, ":") {
+		dnsServer += ":53"
+	}
+
+	return NewClient(dialer, dnsServer)
+}
+
+func NewClient(dialer *net.Dialer, dnsServer string) (*Client, error) {
+	// DNS client
+	client := &dns.Client{
+		Net:          "udp",
+		Dialer:       dialer,
+		DialTimeout:  DialTimeout,
+		ReadTimeout:  DNSReadTimeout,
+		WriteTimeout: DNSWriteTimeout,
+	}
+
+	c := &Client{
```

Comment:
`c` and `client` variables :) In this situation you can embedded `client` inside the struct definition.

And `error` is not used at all.


```
return &Client{
  dnsServer: dnsServer,
  client: &dns.Client{
	Net:          "udp",
	Dialer:       dialer,
	DialTimeout:  DialTimeout,
	ReadTimeout:  DNSReadTimeout,
	WriteTimeout: DNSWriteTimeout, 
  },
}
```